core/shared/src/main/scala/zio/IsSubtype.scala

```diff
@@ -0,0 +1,27 @@
+package zio
+
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("\nOutput Type Mismatch\n  expected: ${B}\n    actual: ${A}\n\n")
+sealed abstract class HasOutput[-A, +B] extends (A => B)
+object HasOutput {
+  implicit def impl[A, B](implicit subtype: A <:< B): HasOutput[A, B] = new HasOutput[A, B] {
+    override def apply(a: A): B = subtype(a)
+  }
+
+  implicit def implNothing[B]: HasOutput[Nothing, B] = new HasOutput[Nothing, B] {
```

Comment:
Do you need these second instances?

Reply:
I know right? Scala weirdness. I did! It couldn't just find one for `Nothing`, which means it didn't trigger the `CanFail` warning, but instead said gave the implicit error for `HasError`.